src/rpc/blockchain.cpp

```diff
@@ -1680,19 +1686,25 @@ static RPCHelpMan getchaintxstats()
 
     const CBlockIndex& past_block{*CHECK_NONFATAL(pindex->GetAncestor(pindex->nHeight - blockcount))};
     const int64_t nTimeDiff{pindex->GetMedianTimePast() - past_block.GetMedianTimePast()};
-    const int nTxDiff = pindex->nChainTx - past_block.nChainTx;
+    const auto window_tx_count{
+        (pindex->nChainTx != 0 && past_block.nChainTx != 0) ? std::optional{pindex->nChainTx - past_block.nChainTx} : std::nullopt,
+    };
 
     UniValue ret(UniValue::VOBJ);
     ret.pushKV("time", (int64_t)pindex->nTime);
-    ret.pushKV("txcount", (int64_t)pindex->nChainTx);
+    if (pindex->nChainTx) {
+        ret.pushKV("txcount", pindex->nChainTx);
+    }
     ret.pushKV("window_final_block_hash", pindex->GetBlockHash().GetHex());
     ret.pushKV("window_final_block_height", pindex->nHeight);
     ret.pushKV("window_block_count", blockcount);
     if (blockcount > 0) {
-        ret.pushKV("window_tx_count", nTxDiff);
+        if (window_tx_count) {
+            ret.pushKV("window_tx_count", *window_tx_count);
+        }
         ret.pushKV("window_interval", nTimeDiff);
-        if (nTimeDiff > 0) {
-            ret.pushKV("txrate", ((double)nTxDiff) / nTimeDiff);
+        if (nTimeDiff > 0 && window_tx_count) {
+            ret.pushKV("txrate", double(*window_tx_count) / nTimeDiff);
```

Comment:
```
rpc/blockchain.cpp: In function ‘getchaintxstats()::<lambda(const RPCHelpMan&, const JSONRPCRequest&)>’:
rpc/blockchain.cpp:1707:34: warning: ‘*(std::_Optional_payload_base<unsigned int>::_Storage<unsigned int, true>*)((char*)&window_tx_count + offsetof(const std::optional<unsigned int>,std::optional<unsigned int>::<unnamed>.std::_Optional_base<unsigned int, true, true>::<unnamed>)).std::_Optional_payload_base<unsigned int>::_Storage<unsigned int, true>::_M_value’ may be used uninitialized [-Wmaybe-uninitialized]
 1707 |             ret.pushKV("txrate", double(*window_tx_count) / nTimeDiff);
      |                                  ^~~~~~~~~~~~~~~~~~~~~~~~
rpc/blockchain.cpp:1689:16: note: ‘*(std::_Optional_payload_base<unsigned int>::_Storage<unsigned int, true>*)((char*)&window_tx_count + offsetof(const std::optional<unsigned int>,std::optional<unsigned int>::<unnamed>.std::_Optional_base<unsigned int, true, true>::<unnamed>)).std::_Optional_payload_base<unsigned int>::_Storage<unsigned int, true>::_M_value’ was declared here
 1689 |     const auto window_tx_count{
      |                ^~~~~~~~~~~~~~~
```

Ubuntu 23.10 with gcc 13.2.0

Which seems spurious, but it goes away with e.g.:

```cpp
    if (blockcount > 0) {
        if (window_tx_count) {
            ret.pushKV("window_tx_count", *window_tx_count);
            if (nTimeDiff > 0) {
                ret.pushKV("txrate", double(*window_tx_count) / nTimeDiff);
            }
        }
        ret.pushKV("window_interval", nTimeDiff);
    }
```

Reply:
> Not sure about changing the code for a broken compiler warning. 

It's also slightly more readable because it checks `window_tx_count` only once.

> I am sure this warning already exists in other parts of the code

This is the only warning thrown.